### PR TITLE
Updates for nautobot 2.0

### PR DESCRIPTION
**Pull request by Dav-C**
Updates for Nautobot 2.0 compatability.

### CODE REVIEW

File: development/Dockerfile [hunk 1]
```diff
@@ -19,20 +19,20 @@ FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
...
-ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
+ENV NAUTOBOT_ROOT=${NAUTOBOT_ROOT}
 
 # Install Poetry manually via its installer script;
 # We might be using an older version of Nautobot that includes an older version of Poetry
 # and CI and local development may have a newer version of Poetry
 # Since this is only used for development and we don't ship this container, pinning Poetry back is not expressly necessary
 # We also don't need virtual environments in container
-RUN curl -sSL https://install.python-poetry.org | python3 - && \
+RUN which poetry || curl -sSL https://install.python-poetry.org | python3 - && \
```
Comment (jvanderaa) [inline development/Dockerfile:29]: I don't understand this change.

File: development/docker-compose.base.yml [hunk 2]
```diff
@@ -21,8 +21,9 @@ services:
         condition: "service_started"
       db:
         condition: "service_healthy"
-    <<: *nautobot-build
-    <<: *nautobot-base
+    <<:
+      - *nautobot-base
+      - *nautobot-build
```
Comment (jvanderaa) [inline development/docker-compose.base.yml:26]:
```suggestion
    <<: [*nautobot-base, *nautobot-build]
```

Comment (jvanderaa) [reply development/docker-compose.base.yml:26]: This is what most other projects have.

File: development/docker-compose.base.yml [hunk 3]
```diff
@@ -37,3 +38,13 @@ services:
...
+  beat:
+    entrypoint:
+      - "sh"
+      - "-c"  # this is to evaluate the $NAUTOBOT_LOG_LEVEL from the env
+      - "nautobot-server celery beat -l $$NAUTOBOT_LOG_LEVEL"  ## $$ because of docker-compose
+    depends_on:
+      - "nautobot"
+    healthcheck:
+      disable: true
+    <<: *nautobot-base
```
Comment (jvanderaa) [inline development/docker-compose.base.yml:50]: What are we doing with this in the migration? I don't think there are any scheduled jobs.

**Review by jvanderaa — CHANGES_REQUESTED** [review]

File: development/Dockerfile (hunk 1 above)
Comment (Dav-C) [reply development/Dockerfile:29]: I used SSoT as a model for changes to the Dockerfile and compose files
https://github.com/nautobot/nautobot-plugin-ssot/blob/35803947ddd8f5c44bf9bbf0df2a91398e9c5539/development/Dockerfile#L30

File: development/docker-compose.base.yml (hunk 3 above)
Comment (Dav-C) [reply development/docker-compose.base.yml:50]: Same as above, I used SSoT as a model for these files. It should probably be updated to `"nautobot-server celery worker -l $$NAUTOBOT_LOG_LEVEL --events" ` to be in line with other plugins.

File: .coveragerc [hunk 4]
```diff
@@ -0,0 +1,7 @@
+[run]
+include =
+    welcome_wizard/*
+    *migrations*
+omit =
+     welcome_wizard/tests/*
+    /usr/*
```
Comment (jvanderaa) [inline .coveragerc:7]: Remove please.

File: development/docker-compose.base.yml (hunk 2 above)
Comment (Dav-C) [reply development/docker-compose.base.yml:26]: nornir, ssot-arista-cloudvision, ssot-servicenow and capacity metrics use `[*nautobot-base, *nautobot-build]` while all the rest use what is already in the PR.

**Review by jvanderaa — APPROVED** [review]

**Comment by whitej6** [comment]
Commits added in another PR

**Pull request closed without merging**